core/src/tile/tileManager.cpp

```diff
@@ -384,11 +388,24 @@ void TileManager::loadSubTasks(std::vector<std::shared_ptr<DataSource>>& _subSou
         if (it != subTasks.end() && (*it)->subTaskId() == int(index)) { continue; }
 
         TileID subTileID = tileID;
+
+        // get tile for lower zoom if we are past max zoom
         if (subTileID.z > subSource->maxZoom()) {
             subTileID = subTileID.withMaxSourceZoom(subSource->maxZoom());
         }
         auto subTask = subSource->createTask(subTileID, index);
-        if (subTask->isReady()) {
+        // check if we are at valid zoom for source
+        if (!subSource->isActiveForZoom((float)subTileID.z)) {
+            // Right now all subSources are raster, but let's check anyway...
+            if (RasterSource* rasterSubSource = dynamic_cast<RasterSource*>(subSource.get())) {
+                subTasks.insert(it, subTask);
+                // If the subSource isn't active for the zoom, we should just
+                // load an empty texture and move on.
+                rasterSubSource->loadEmptyTexture(std::move(subTask));
```

Comment:
When I was implementing this, I was trying things out with Walkabout. My thought was that our normals may not cover a given layer. For example, what if we had normals of zooms 8 - 12, but we wanted to render vectors with 0 - 18? It looks like it works fine as-is--the hillshades just don't get drawn when the sub source doesn't load.